### ISSUE TITLE
Could not receive from subscribed topic. UPD: at MQTT v.5

### ISSUE THREAD

**Issue by CHE77** [flespi-software/MQTT-Tiles#36]
Used HiveMQ broker
Publishing works. - Got the messages at web client. 
Could not receive from subscribed topic.
![data (2).txt](https://github.com/flespi-software/MQTT-Tiles/files/13963637/data.2.txt)

![Screenshot_2024-01-17-09-30-08-313_com.android.chrome.jpg](https://github.com/flespi-software/MQTT-Tiles/assets/4175310/efcad007-5c36-412c-97b1-ddb99cf43b09)

**Comment by CHE77** [comment]
Problem was at slow internet, big ping. Some mqtt brokers do not pass messages if ping > 700ms. But they show connection as green. 
It would be nice to have widget with ping to broker or other quality of connection indication.
It looks like this blogger also failed to show basic mqtt actions 
https://www.youtube.com/watch?v=_4zVNPEZI8g&t=234

**Comment by dexif** [comment]
Now you can see the ping. The ping packet is sent once every 2*keep_alive.
![screenshot-mqtttiles flespi io-2024 01 18-15_27_37](https://github.com/flespi-software/MQTT-Tiles/assets/387234/6be3fe57-8fbc-4e42-a957-6f067e20a406)

**Comment by bam80** [comment]
@CHE77 maybe rising QoS wold help in your case?
IIRC, "0" by default means you will receive the packets _at most_ once.

**Comment by CHE77** [comment]
> @CHE77 maybe rising QoS would help in your case?
Tested with different QoS - no changes.
In my case ping - 700ms, due to satellite connection. To make WSS connection with broker is more difficult than TCP. But when it done on IotManager app publishing and receiving work fine, and at MQTT Tiles - only for publishing. At same broker - clusterfly.ru mqtt v5
Must say at MQTT Tiles could receive messages only via live-control.com broker v3.1
But have some other issues there

It is good to have possibility to set QoS at MQTT Tiles as well.

**Comment by bam80** [comment]
I have no idea how it can happen:
after all, it's TCP - so the packets should be guaranteed to be delivered as long as TCP connection is not broken, isn't it @dexif ?

**Comment by CHE77** [comment]
> @CHE77 maybe rising QoS would help in your case?
Tested with different QoS - no changes.
In my case ping - 700ms, due to satellite connection. To make WSS connection with broker is more difficult than TCP. But when it done on IotManager app publishing and receiving work fine, and at MQTT Tiles - only for publishing. At same broker - clusterfly.ru mqtt v5
Must say at MQTT Tiles could receive messages only via live-control.com broker v3.1
But have some other issues there

It is good to have possibility to set QoS at MQTT Tiles as well.

**Comment by CHE77** [comment]
> @CHE77 maybe rising QoS would help in your case?
Tested with different QoS - no changes.
In my case ping - 700ms, due to satellite connection. To make WSS connection with broker is more difficult than TCP. But when it done on IotManager app publishing and receiving work fine, and at MQTT Tiles - only for publishing. At same broker - clusterfly.ru mqtt v5
Must say at MQTT Tiles could receive messages only via live-control.com broker v3.1
But have some other issues there

It is good to have possibility to set QoS at MQTT Tiles as well. 

UPD: I changed all connections to MQTT v.3.1.1 and now all ok, at any ping)
But clusterfly.ru and HiveMQ support v.5. Could not make them work at v.5 even with good ping
